### ISSUE TITLE
Cannot activate Python extension

### ISSUE THREAD

**Issue by adamjstewart** [LLNL/spack#1034]
Is there any way to specify multiple version ranges for dependencies? For example, Python modules commonly depend on Python 2.6, 2.7, or 3.4+

**Comment by tgamblin** [comment]
`depends_on('python@2.6:2.7,3.4:')`

**Comment by adamjstewart** [comment]
No way, that's awesome. I was actually just about to suggest that exact same syntax if we didn't have that ability yet!

**Comment by adamjstewart** [comment]
Reopening because I have a new problem. In my revamped py-matplotlib, I have:
```python
extends('python@2.7:2.8,3.4:3.5', ignore=r'bin/nosetests.*$|bin/pbr$')
```
However, when I try to activate the package I installed, it says:
```
==> Error: py-matplotlib is not an extension.
```
If I use a more reasonable extension, like:
```python
extends('python', ignore=r'bin/nosetests.*$|bin/pbr$')
```
I get:
```
==> Error: Spec oalinbvqe3ant7u3ox2gxmuyan5lvxjx not found in /blues/gpfs/home/software/spack/opt/spack/linux-x86_64/gcc-6.1.0/py-astropy-1.1.2-oalinbvqe3ant7u3ox2gxmuyan5lvxjx
```
I don't have py-astropy installed, nor is it a dependency of py-matplotlib. I was working on installing it earlier. Could it have stayed in my PYTHONPATH somehow?

**Comment by adamjstewart** [comment]
I'm looking at [this section of code](https://github.com/LLNL/spack/blob/develop/lib/spack/spack/package.py#L587) and I have a hunch that it's causing the problem. If I remember correctly, someone just edited this. Not sure if it was already merged or not. Can anyone point me to the PR?

**Comment by adamjstewart** [comment]
I have a new theory. Some of the PYTHONPATH bugs may have occurred because I was using a Spack-installed Python to run Spack to build packages for that very same Python. I guess it shouldn't be surprising that it got confused. I'm going to try to uninstall and reinstall everything to see if I can successfully activate Python packages.

**Comment by adamjstewart** [comment]
The error message:
```
==> Error: Spec oalinbvqe3ant7u3ox2gxmuyan5lvxjx not found in /blues/gpfs/home/software/spack/opt/spack/linux-x86_64/gcc-6.1.0/py-astropy-1.1.2-oalinbvqe3ant7u3ox2gxmuyan5lvxjx
```
was due to the fact that I had py-astropy in my Python extensions.yaml. I must have manually deleted py-astropy without deactivating it first.

**Comment by adamjstewart** [comment]
Alright, everything is finally copacetic. I was trying to install with the same Python package that I was installing to, so that was a problem. I also had never-ending trouble when trying to specify the versions for Python extensions, so I stopped doing that. I also accidentally deleted the py-astropy installation without deactivating it first, so I had to delete the binaries, egg, and remove it from extensions.yaml.

I have no idea which of these things was causing which problem, but I also don't care anymore. It's Friday :smiley:
